snippets/base/models.py

```diff
@@ -667,6 +671,34 @@ def get_admin_url(self):
         full_url = urljoin(settings.SITE_URL, url)
         return full_url
 
+    def duplicate(self, creator):
+        snippet_copy = copy.deepcopy(self)
+        snippet_copy.id = None
+        snippet_copy.created = None
+        snippet_copy.modified = None
+        snippet_copy.status = STATUS_CHOICES['Draft']
+        snippet_copy.creator = creator
+        snippet_copy.uuid = uuid.uuid4()
+        snippet_copy.name = '{0} - {1}'.format(
+            self.name,
+            datetime.strftime(datetime.now(), '%Y.%m.%d %H:%M:%S'))
+        snippet_copy.save()
+
+        for field in self._meta.get_fields():
+            attr = getattr(self, field.name)
+            if isinstance(attr, Manager):
+                manager = attr
+                if manager.__class__.__name__ == 'RelatedManager':
+                    for itm in manager.all():
+                        itm_copy = copy.copy(itm)
+                        itm_copy.id = None
+                        getattr(snippet_copy, field.name).add(itm_copy)
+                elif manager.__class__.__name__ == 'ManyRelatedManager':
+                    for snippet in manager.all():
+                        getattr(snippet_copy, field.name).add(snippet)
```

Comment:
If it's feasible, it would be nice to have tests for these related items.

Reply:
Added another test for `ManyRelatedManager`. We don't have any `RelatedManager`s in ASRSnippet yet so cannot test the other.